File: backlog/src/main/java/com/agilefox/backlog/model/State.java
package com.agilefox.backlog.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Data
@Entity
@Table(name = "state")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class State {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;

    private String description;

    private long projectId;

    @ManyToOne
    @JoinColumn( name="typeId", nullable = false)
    private Type type;

    private int stateOrder;


}

File: agilefox-frontend/src/types/State.ts
import { z } from "zod";

export const StateSchema = z.object({
  id: z.number(),
  name: z.string(),
  description: z.string().optional(),
  projectId: z.number(),
  stateOrder: z.number(),
  typeId: z.number(),
});

export type State = z.infer<typeof StateSchema>;
